lib/temporal/client.rb

```diff
@@ -9,15 +9,50 @@
 
 module Temporal
   class Client
+    # @return [String] Namespace used for client calls.
     attr_reader :namespace
 
-    # TODO: More argument to follow for converters, codecs, etc
+    # Create a Temporal client from a service client.
+    #
+    # @param connection [Temporal::Connection] A previously established connection with the
+    #   Temporal server.
+    # @param namespace [String] Namespace to use for client calls.
+    # @param interceptors [Array<Temporal::Interceptor::Client>] List of interceptors for
+    #   intercepting client calls. Executed in their original order.
     def initialize(connection, namespace, interceptors: [])
       @namespace = namespace
       data_converter = DataConverter.new
```

Comment:
Shouldn't this accept a data converter too?

Reply:
Yep, this will accept `payload_converter`, `payload_codecs` and `failure_converter`. Will add in a separate PR